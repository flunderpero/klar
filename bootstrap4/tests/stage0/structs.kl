struct Point:
    x i32
    y i32
end

struct Line:
    start Point
    end_ Point
end

fn simple():
    let p = Point(1, 2)
    print(p.x)
    print(p.y)
end

fn mutable():
    mut mp = Point(3, 4)
    mp.x = 5
    mp.y = mp.x + 1
    print(mp.x)
    print(mp.y)
end

fn nested():
    mut line = Line(Point(10, 11), Point(12, 13))
    line.end_.x = 16
    line.end_.y = line.end_.x + 1
    print(line.start.x)
    print(line.start.y)
    print(line.end_.x)
    print(line.end_.y)
end


fn main():
    simple()
    mutable()
    nested()
    return 0
end
