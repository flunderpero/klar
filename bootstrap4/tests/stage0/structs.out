1
2
5
6
10
11
16
17
---
exit code: 0
